conan/tools/env/environment.py

```diff
@@ -536,7 +544,14 @@ def save_script(self, filename):
             is_ps1 = ext == ".ps1"
         else:  # Need to deduce it automatically
             is_bat = self._subsystem == WINDOWS
-            is_ps1 = self._conanfile.conf.get("tools.env.virtualenv:powershell", check_type=bool)
+            try:
+                is_ps1 = self._conanfile.conf.get("tools.env.virtualenv:powershell", check_type=bool)
+                ConanOutput().warning(
+                    "Boolean values for 'tools.env.virtualenv:powershell' are deprecated. "
+                    "Please use 'powershell' or 'pwsh'.", warn_tag="deprecated"
```

Comment:
How would a post-deprecation disabling of ps1 work in cases where

- It's set by default for all packages in the profile
- But for specific packages, you still want to disable it?

Do we expect users to discover

```
tools.env.virtualenv:powershell=powershell
my-bat-only-package/*:tools.env.virtualenv:powershell=!
```

Reply:
Yes, I think we could use the `=!` ? Or setting it to an empty value to make the same thing as `False`.